maestro-cli/src/main/java/maestro/cli/device/DeviceCreateUtil.kt

```diff
@@ -103,11 +105,11 @@ internal object DeviceCreateUtil {
         }
 
         val systemImage = config.systemImage
-        val deviceName = config.deviceName
+        var deviceName = config.deviceName + shardIndex?.let { "_${it + 1}" }
 
         // check connected device
-        if (DeviceService.isDeviceConnected(deviceName, Platform.ANDROID) != null) {
-            throw CliError("A device with name $deviceName is already connected")
+        while (DeviceService.isDeviceConnected(deviceName, Platform.ANDROID) != null) {
+            deviceName += "_"
```

Comment:
Did not follow here, why adding an _ to the name of the device would make this work?  
Maybe related to this line: https://github.com/mobile-dev-inc/maestro/pull/1732/files#diff-cb7f7f66181dc16c2226f299c8649e8f30be45c7c13deeda207d0d255cb9c61cR30?

Reply:
you're right, I was taking care of this part around there, where you included the link. 

But I vaguely remember running 2 maestro instances with multiple shards or some weird setup like that, and the other maestro session's first shard had the exact same pixel name, and this part of the code prevented me from running a 2nd maestro with 2 shards. So one with 2 shards, another with 2 shards, second one ends up trying to create device with same name and causes to fail. But this quick workaround lets you create duplicates of shards as well. It might be redundant, maybe sharding is enough by itself :D I'll remove this if you want ?